src/network/networkd-link.c

```diff
@@ -1490,6 +1490,30 @@ bool link_has_carrier(Link *link) {
         return false;
 }
 
+bool link_should_activate(Link *link) {
+        assert(link);
+
+        if (!link->network)
+                return true;
+
+        if (link->network->activation_mode == ACTIVATION_MODE_OFF)
+                return false;
+
+        return true;
+}
+
+bool link_is_manual(Link *link) {
+        assert(link);
+
+        if (!link->network)
+                return false;
+
+        if (link->network->activation_mode == ACTIVATION_MODE_MANUAL)
+                return true;
+
+        return false;
+}
```

Comment:
with the suggestion to make this a tri-state (see above) this funciton is probably not necessary?